drivers/stepper/ti/drv8424.c

```diff
@@ -0,0 +1,775 @@
+
+	/* Check availability of sleep and enable pins, as these might be hardwired. */
+	if (!has_sleep_pin && !has_enable_pin) {
+		LOG_ERR("%s: Failed to enable/disable device, neither sleep pin nor enable pin are "
+			"available. The device is always on.",
+			dev->name);
+		return -ENOTSUP;
+	}
+
+	if (has_sleep_pin) {
+		ret = gpio_pin_set_dt(&config->sleep_pin, !enable);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to set sleep_pin (error: %d)", dev->name, ret);
+			return ret;
+		}
+		data->pin_states.sleep = enable ? 0U : 1U;
+	}
+
+	if (has_enable_pin) {
+		ret = gpio_pin_set_dt(&config->en_pin, enable);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to set en_pin (error: %d)", dev->name, ret);
+			return ret;
+		}
+		data->pin_states.en = enable ? 1U : 0U;
+	}
+
+	data->enabled = enable;
+	if (!enable) {
+		counter_stop(config->counter);
+		data->is_moving = false;
+		gpio_pin_set_dt(&config->step_pin, 0);
+		data->step_signal_high = false;
+	}
+
+	return 0;
+}
+
+static int drv8424_set_counter_frequency(const struct device *dev, uint32_t freq_hz)
+{
+	const struct drv8424_config *config = dev->config;
+	struct drv8424_data *data = dev->data;
+	int ret;
+
+	if (freq_hz == 0) {
+		return -EINVAL;
+	}
+
+	data->counter_top_cfg.ticks = counter_us_to_ticks(config->counter, USEC_PER_SEC / freq_hz);
+
+	ret = counter_set_top_value(config->counter, &data->counter_top_cfg);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to set counter top value (error: %d)", dev->name, ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int drv8424_start_positioning(const struct device *dev)
+{
+
+	const struct drv8424_config *config = dev->config;
+	struct drv8424_data *data = dev->data;
+	int ret;
+
+	/* Unset constant velocity flag if present */
+	data->constant_velocity = false;
+
+	/* Set direction pin */
+	int dir_value = (data->target_position >= data->actual_position) ? 1 : 0;
+
+	ret = gpio_pin_set_dt(&config->dir_pin, dir_value);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to set direction pin (error %d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Lock interrupts while modifying counter settings */
+	int key = irq_lock();
+
+	/* Set counter to correct frequency */
+	ret = drv8424_set_counter_frequency(dev, data->max_velocity * 2);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to set counter frequency (error %d)", dev->name, ret);
+		goto end;
+	}
+
+	data->is_moving = true;
+
+	/* Start counter */
+	ret = counter_start(config->counter);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to start counter (error: %d)", dev->name, ret);
+		data->is_moving = false;
+		goto end;
+	}
+
+end:
+	irq_unlock(key);
+	return ret;
+}
+
+static int drv8424_move_by(const struct device *dev, int32_t micro_steps)
+{
+	struct drv8424_data *data = dev->data;
+	int ret;
+
+	if (data->max_velocity == 0) {
+		LOG_ERR("%s: Invalid max. velocity %d configured", dev->name, data->max_velocity);
+		return -EINVAL;
+	}
+
+	if (!data->enabled) {
+		return -ENODEV;
```

Comment:
This check should be there, however there has to be another errno. Probably [ECANCELLED](https://docs.zephyrproject.org/apidoc/latest/group__system__errno.html#ga9532d840ef91fd8e1ecc5d7ca7cbf212) would be a more suitable option.